x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/type/DataType.java

```diff
@@ -36,19 +36,19 @@
     DOUBLE(        "double",         JDBCType.DOUBLE,    Double.BYTES,      15,                25, false, true,  true),
     // 24 bits defaultPrecision - 24*log10(2) =~ 7 (7.22)
     FLOAT(         "float",          JDBCType.REAL,      Float.BYTES,       7,                 15, false, true,  true),
-    HALF_FLOAT(    "half_float",     JDBCType.FLOAT,     Double.BYTES,      16,                25, false, true,  true),
+    HALF_FLOAT(    "half_float",     JDBCType.FLOAT,     Float.BYTES,       3,                 25, false, true,  true),
     // precision is based on long
-    SCALED_FLOAT(  "scaled_float",   JDBCType.FLOAT,     Double.BYTES,      19,                25, false, true,  true),
-    KEYWORD(       "keyword",        JDBCType.VARCHAR,   Integer.MAX_VALUE, 256,               0,  false, false, true),
-    TEXT(          "text",           JDBCType.VARCHAR,   Integer.MAX_VALUE, Integer.MAX_VALUE, 0,  false, false, false),
+    SCALED_FLOAT(  "scaled_float",   JDBCType.DOUBLE,    Integer.BYTES,     15,                25, false, true,  true),
+    KEYWORD(       "keyword",        JDBCType.VARCHAR,   Integer.MAX_VALUE, 256,               256,false, false, true),
```

Comment:
I wonder if `256` is correct here. I think that's the default, but IIRC, this is a setting and thus could have a different (maximum) value. Treating `KEYWORD` as `TEXT` in this regard might be potentially safer?

It otherwise LGTM.

Reply:
I put 256 since I thought it was the default `ignore_above` value but I might be wrong:
https://www.elastic.co/guide/en/elasticsearch/reference/current/ignore-above.html
You are right that it can be larger however I'm not sure what the default value should be - it's significantly smaller than that of `text` - seems like `Short.MAX_VALUE-1` (32766) is the max.